tests/forward_diffusion_test.py

```diff
@@ -0,0 +1,72 @@
+import torch
+import seaborn as sns
+import matplotlib.pyplot as plt
+from torch.utils.data import DataLoader
+
+from ecg_segmentation_dataset import ECGDataset
+from models.forward_diffusion import ForwardDiffusion
+
+
+def preprocess_dataset(dataset_name: str, batch_size: int, num_workers: int):
+    train_ds = ECGDataset(dataset_name, split="train")
+    val_ds = ECGDataset(dataset_name, split="validation")
+    test_ds = ECGDataset(dataset_name, split="test")
+
+    # dataloaders
+    train_dataloader = DataLoader(train_ds, batch_size=batch_size, num_workers=num_workers)
+    val_dataloader = DataLoader(val_ds, batch_size=batch_size, num_workers=num_workers)
+    test_dataloader = DataLoader(test_ds, batch_size=batch_size, num_workers=num_workers)
+
+    return train_dataloader, val_dataloader, test_dataloader
+
+
+if __name__ == "__main__":
+    # load data
+    _, _, test_dataloader = preprocess_dataset("roszcz/ecg-segmentation-ltafdb", 4, 1)
+
+    # initialing random input
+    records = next(iter(test_dataloader))
+    signal = records["signal"]
+
+    schedule_type = "sigmoid"
+
+    # initialize forward diffusion
+    fdiff = ForwardDiffusion(beta_start=0.0001, beta_end=0.02, timesteps=256, schedule_type=schedule_type)
+
+    # timesteps for visualization
+    t = torch.tensor([0, 63, 127, 191, 255], dtype=torch.long)
```

Comment:
```suggestion
    t = torch.tensor([0, 10, 30, 63, 127, 255], dtype=torch.long)
```

63 steps seems like a lot, can you post some images for the earlier steps as well?

Reply:
![forward_diffusion_cosine_heartbeats](https://github.com/Nospoko/ecg-diffusion/assets/94491798/d31a757e-3aa2-4ebc-b8c3-10c77deb595b)
![forward_diffusion_sigmoid_heartbeats](https://github.com/Nospoko/ecg-diffusion/assets/94491798/b7cc4006-8c8f-4506-990d-3955e64e3c5d)